src/update.c

```diff
@@ -278,20 +349,57 @@ int main_update()
 		goto clean_curl;
 	}
 
+	mix_exists = check_mix_exists();
+
 	fprintf(stderr, "Update started.\n");
 
 	grabtime_start(&times, "Update Step 1: get versions");
 
-	read_subscriptions_alt(&current_subs);
-
-	/* Step 1: get versions */
+	read_subscriptions_alt(&current_subs, mix_exists);
 
+/* Step 1: get versions */
+version_check:
 	ret = check_versions(&current_version, &server_version, path_prefix);
 
 	if (ret < 0) {
 		ret = EXIT_FAILURE;
 		goto clean_curl;
 	}
+
+	if (mix_exists) {
+		ret = check_mix_versions(&mix_current_version, &mix_server_version, path_prefix);
+		if (ret < 0) {
+			ret = EXIT_FAILURE;
+			goto clean_curl;
+		}
+		/* Check if a new upstream version is available so we can update to it still */
+		if (need_new_upstream(server_version)) {
+			printf("NEW CLEAR AVAILABLE %d\n", server_version);
```

Comment:
debug print

Reply:
This was an intended print, so does everyone agree to remove it?

Reply:
Because the message is in all-caps, it caught my attention. If that's intended, then okay.

Reply:
This seems useful to me.

Reply:
Okay, +1 for this line.